src/wast-lexer.cc

```diff
@@ -32,27 +32,77 @@
 
 #define INITIAL_LEXER_BUFFER_SIZE (64 * 1024)
 
-#define YY_USER_ACTION                  \
-  {                                     \
-    loc->filename = filename_;          \
-    loc->line = line_;                  \
-    loc->first_column = COLUMN(token_); \
-    loc->last_column = COLUMN(cursor_); \
-  }
+#define PUSH_LOOKAHEAD_TOKEN              \
+  do {                                    \
+    WastLexer::LexToken tok;              \
+    lookahead_->tokens_.push_back(tok);   \
+    token_ = &lookahead_->tokens_.back(); \
+  } while (0)
+
+#define YY_USER_ACTION(loc)                \
```

Comment:
There's nothing magical about YY_USER_ACTION name anymore, probably better to give it a clearer name like SET_LOCATION or FILL_LOCATION or something.